### PR TITLE
PO-1991 - Get Minor Creditor Account Legacy

### PR DESCRIPTION
**Pull request by Aneesah33**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PO-1991

### Change description ###

- Adds legacy-mode support for GET /minor-creditor-accounts/{id}
- Wires the endpoint through controller, shared service, proxy, and legacy service
- Uses legacy gateway actionType GET_MINOR_CREDITOR_ACCOUNT_PARTY with accountId request mapping
- Maps legacy response DTOs into the API response using legacy-specific mappers
- Applies Search and View Accounts auth and filters BACS details unless the user has View Creditor BACS in the account business unit
- Leaves Opal mode unsupported in this ticket
- Adds test coverage for controller/service/legacy service and legacy integration scenarios
- Adds matching legacy stub mapping and XML response in opal-legacy-db-stub

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/opal/controllers/MinorCreditorApiController.java [hunk 1]
```diff
@@ -27,6 +27,18 @@ public Optional<NativeWebRequest> getRequest() {
         return Optional.ofNullable(request);
     }
```
Comment (Sean-Reed-CGI) [inline src/main/java/uk/gov/hmcts/opal/controllers/MinorCreditorApiController.java:25]: Looks like the use of NativeWebRequest stored as a field in this controller wouldn't be thread-safe

Comment (Aneesah33) [reply src/main/java/uk/gov/hmcts/opal/controllers/MinorCreditorApiController.java:25]: Agreed, I spoke to Ben and we aligned on using Spring Security context for this path. I’ve removed the NativeWebRequest handling from the controller, so it no longer stores request-specific state, and the authenticated user is now resolved via UserStateService from Spring Security context.

**Review by DeclanClarkeCGI — COMMENTED** [review]
The API has been designed as Get Minor Creditor Account but we have some references to Get Minor Creditor Account Party, here and in the legacy stub. I appreciate there are party details in the response but it would be good if we can be consistent.

**Review by DeclanClarkeCGI — APPROVED** [review]

**Comment by Sean-Reed-CGI** [comment]
Please make sure all new unit tests have the // Arrange // Act // Assert comments

File: src/test/java/uk/gov/hmcts/opal/service/UserStateServiceTest.java [hunk 2]
```diff
@@ -52,6 +55,24 @@ void testCheckForAuthorisedUser_success() {
         assertEquals(1L, userState.getUserId());
     }
```
Comment (Sean-Reed-CGI) [inline src/test/java/uk/gov/hmcts/opal/service/UserStateServiceTest.java]: Missing unit tests for authorization==null and when principal is not a Jwt

Comment (HennaParmar) [reply src/test/java/uk/gov/hmcts/opal/service/UserStateServiceTest.java]: Added both missing cases in `UserStateServiceTest`:
- `authorization == null`
- security context principal is not a `Jwt`

This now covers the new `SecurityContext` path and the null-authorization fallback explicitly.

**Review by Sean-Reed-CGI — APPROVED** [review]
LGTM

**Review by CadeFaulkner — APPROVED** [review]
